### PR TITLE
[NetAppFiles] update to 2023-11-01

### PR DESCRIPTION
**Pull request by audunn**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by NoriZC** [comment]
Hi @audunn please update the changelog and also check the box to illustrate what kind of release are you proposing:
image

**Comment by audunn** [comment]
@NoriZC  PR updated

File: src/NetAppFiles/NetAppFiles/ChangeLog.md [hunk 1]
```diff
@@ -18,6 +18,26 @@
...
+    * Paramters `PoolName` and 'VolumeName' are no longer mandatory, not needed and will be deprecated in Backup cmdlets
+        - `New-AzNetAppFilesBackup`, `Get-AzNetAppFilesBackup`, `Update-AzNetAppFilesBackup`, `Remove-AzNetAppFilesBackup` and `Restore-AzNetAppFilesBackupFile`
+    * Added Property `VolumeResourceId ` to PSNetAppFilesBackup
+        - `New-AzNetAppFilesBackup`, `Get-AzNetAppFilesBackup`, `Update-AzNetAppFilesBackup`
+    * Added Paramter `BackupVaultId` to `Backup`. object 
+        - `New-AzNetAppFilesVolume`, `Update-AzNetAppFilesVolume`
+    * Added Paramter `ProtocolType` to `Update-AzNetAppFilesVolume`
+    * Updated contraints on Paramter `CoolnessPeriod`  minimum 2, maximum 183
+        - `New-AzNetAppFilesVolume`, `Update-AzNetAppFilesVolume`
+    * Updated help description for parameters `SnaphotId` and `BackupId` to clarify they should now only be used with backup and snapshot resourceId respectively in `New-AzNetAppFilesVolume`
```
Comment (NoriZC) [inline src/NetAppFiles/NetAppFiles/ChangeLog.md:40]:
```suggestion
## Upcoming Release
* Updated to api-version 2023-11-01
```

The changelog is for customer ,different from syntax changelog. And hopefully we can use simple words to explain the change.

**Comment by NoriZC** [comment]
@audunn Kindly reminder. Please help confirm and commit the suggested updates.

**Comment by NoriZC** [comment]
> @NoriZC Updated changelog. Not exactly sure what changes are needed. Please let me know if more changes are needed and what.

As I mentioned before, we only need a line mention the api upgrade in our changelog as suggested:

image

**Review by NoriZC — APPROVED** [review]

**Pull request merged**